internal/shardchain/dbfactory.go

```diff
@@ -67,3 +122,8 @@ func (f *LDBShardFactory) NewChainDB(shardID uint32) (ethdb.Database, error) {
 		CacheSize: f.CacheSize,
 	})), nil
 }
+
+// MmrDB returns OpenMmrDB for the blockchain for given shard and epoch.
+func (f *LDBShardFactory) MmrDB(shardID uint32) (core.OpenMmrDB, error) {
```

Comment:
Is the reason to have a separate db for mmr data that it have potential conflict with the chain id entries?

Reply:
no, the main reason is to use the database implementation of the MMR library directly, it simply stores/loads all nodes to/from a file. and also MMR file DB is optional, we can rebuild it from blocks of an epoch.